### PR TITLE
[AAP-8279][AAP-8313] Update credential type selection on credential form

### PR DESCRIPTION
**Pull request by nixocio**
Update credential type selection on credential form

image

### CODE REVIEW

File: frontend/awx/access/credentials/CredentialForm.tsx [hunk 1]
```diff
@@ -249,31 +249,23 @@ function CredentialInputs({ isEditMode = false }: { isEditMode?: boolean }) {
...
+      />
       <PageFormSelectOrganization<Credential> isRequired={isGalaxyCredential} name="organization" />
-      <PageFormSection singleColumn>
-        <PageFormTextArea<Credential>
-          name="description"
-          label={t('Description')}
-          placeholder={t('Enter description')}
-        />
-      </PageFormSection>
+      <PageFormSelectCredentialType<Credential>
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialForm.tsx:258]: Should the interface here be `CredentialType`?

File: frontend/awx/access/credentials/components/PageFormSelectCredentialType.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { useCredentialTypesFilters } from '../../credential-types/hooks/useCredentialTypesFilters';
+import { useCredentialTypesColumns } from '../../credential-types/hooks/useCredentialTypesColumns';
+import { CredentialType } from '../../../interfaces/CredentialType';
+
+/**
+ * A form input for selecting an credential type.
+ *
+ * @example
+ * ```tsx
+ * <PageFormSelectCredentialType<Credential> name="organization" />
```
Comment (kialam) [inline frontend/awx/access/credentials/components/PageFormSelectCredentialType.tsx:14]:
```suggestion
 * <PageFormSelectCredentialType<CredentialType> name="credential-type" />
```

File: frontend/awx/access/organizations/components/PageFormOrganizationSelect.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { useOrganizationsColumns, useOrganizationsFilters } from '../Organizatio
  *
  * @example
  * ```tsx
- * <PageFormSelectOrganization<Credential> name="organization" />
+ * <PageFormSelectOrganization<Organization> name="organization" />
```
Comment (kialam) [inline frontend/awx/access/organizations/components/PageFormOrganizationSelect.tsx:13]: 👍

File: frontend/awx/access/credentials/CredentialForm.tsx (hunk 1 above)
Comment (nixocio) [reply frontend/awx/access/credentials/CredentialForm.tsx:258]: Yes, good catch.

**Pull request merged**